app/classes/DB.php

```diff
@@ -252,25 +265,33 @@ public function sql2db(string $query, array $params = []) {
      *
      * @param string $query The SQL query.
      * @param array $params Parameters for the SQL query (optional).
-     * @return string The result of the SQL query as a JSON-encoded string.
+     * @return string|bool The result of the SQL query as a JSON-encoded string.
+     * @throws Exception
      */
-    public function sql2json(string $query, array $params = []):string {
+    public function sql2json(string $query, array $params = []): string|bool
```

Comment:
Same here, we can't use "string|bool". Union Types is an > PHP 8 thing. PHP 7 only supports 1 return type.